### ISSUE TITLE
Bug: INWX configuration template seems to be wrong

### ISSUE THREAD

**Issue by shnizzl** [qdm12/ddns-updater#410]
**TLDR**: *INWX configuration template seems to be wrong*

1. Is this urgent: No
2. DNS provider(s) you use: INWX
3. Program version:

    

    `Running version latest built on 2022-12-14T20:23:40Z (commit 3fd82ab)`

4. What are you using to run the container: docker-compose
5. Extra information (optional)

The INWX config template seems to be a copy of the OpenDNS config template. My knowledge about all these config parameters is too limited to fix this myself, but I am happy to test things if someone would drop me a few hints. 

 
Removing the **,** in line 9 and using **dyn** as provider the logs are as follows:

```log
2023-01-08T18:32:04+01:00 INFO reading JSON config from file /updater/data/config.json
2023-01-08T18:32:04+01:00 INFO Found single setting to update record
2023-01-08T18:32:04+01:00 INFO Reading history from database: domain ***.*** host ***
2023-01-08T18:32:04+01:00 INFO [healthcheck server] listening on 127.0.0.1:9999
2023-01-08T18:32:04+01:00 INFO [backup] disabled
2023-01-08T18:32:04+01:00 INFO [http server] listening on :8000
2023-01-08T18:32:04+01:00 INFO IPv4 address of ***.***.*** is 127.0.0.1 and your IPv4 address is 84.150.***.***
2023-01-08T18:32:04+01:00 INFO Updating record [domain: ***.*** | host: cloud | provider: Dyn] to use 84.150.***.***
2023-01-08T18:32:05+01:00 ERROR bad request sent
```

Configuration file (**remove your credentials!**):

```json
{
  "settings": [
    {
      "provider": "dyn",
      "domain": "***.***",
      "host": "***",
      "username": "***",
      "password": "***",
      "ip_version": "ipv4"
    }
  ]
}
```

If I use **inwx** as provider it looks good, but the IP is not updated. There is no possibility to check logs on the INWX site. 
```log
2023-01-08T18:40:09+01:00 INFO reading JSON config from file /updater/data/config.json
2023-01-08T18:40:09+01:00 INFO Found single setting to update record
2023-01-08T18:40:10+01:00 INFO Reading history from database: domain ***.*** host ***
2023-01-08T18:40:10+01:00 INFO [healthcheck server] listening on 127.0.0.1:9999
2023-01-08T18:40:10+01:00 INFO [backup] disabled
2023-01-08T18:40:10+01:00 INFO [http server] listening on :8000
2023-01-08T18:40:11+01:00 INFO IPv4 address of ***.***.*** is 80.146.***.*** and your IPv4 address is 84.150.***.***
2023-01-08T18:40:11+01:00 INFO Updating record [domain: ***.*** | host: *** | provider: INWX] to use 84.150.***.***

** Press ANY KEY to close this window ** 
```

```json
{
  "settings": [
    {
      "provider": "inwx",
      "domain": "***.***",
      "host": "***",
      "username": "***",
      "password": "***",
      "ip_version": "ipv4"
    }
  ]
}
```

**Comment by max06** [comment]
I can confirm: 
```log
core-ddns-1  | 2023-01-10T15:41:32Z INFO IPv4 address of $host is $old and your IPv4 address is $new
core-ddns-1  | 2023-01-10T15:41:32Z INFO Updating record [domain: $domain | host: $host | provider: INWX] to use $new
core-ddns-1  | panic: assignment to entry in nil map
core-ddns-1  |
core-ddns-1  | goroutine 60 [running]:
core-ddns-1  | net/url.Values.Set(...)
core-ddns-1  |  net/url/url.go:896
core-ddns-1  | github.com/qdm12/ddns-updater/internal/settings/providers/inwx.(*Provider).Update(0xc000686280, {0x11514f8, 0xc0004faf00}, 0x1?, {0xc0004500a0, 0x10, 0x10})
core-ddns-1  |  github.com/qdm12/ddns-updater/internal/settings/providers/inwx/provider.go:105 +0x27f
core-ddns-1  | github.com/qdm12/ddns-updater/internal/update.(*Updater).Update(0xc0004e3d10, {0x11514f8, 0xc0004faf00}, 0xc0e77fab09440312?, {0xc0004500a0, 0x10, 0x10}, {0xc0e77fab0986a1dd, 0x214093c3, 0x1a24e40})
core-ddns-1  |  github.com/qdm12/ddns-updater/internal/update/update.go:46 +0x22e
core-ddns-1  | github.com/qdm12/ddns-updater/internal/update.(*Runner).updateNecessary(0xc0004e4e10, {0x11514f8, 0xc0004faf00}, {0xc0004f8280, 0x10, 0x10})
core-ddns-1  |  github.com/qdm12/ddns-updater/internal/update/run.go:287 +0x77d
core-ddns-1  | github.com/qdm12/ddns-updater/internal/update.(*Runner).Run(0xc0004e4e10, {0x11514f8, 0xc0004faf00}, 0x1c?)
core-ddns-1  |  github.com/qdm12/ddns-updater/internal/update/run.go:304 +0x15d
core-ddns-1  | created by main._main
core-ddns-1  |  ./main.go:239 +0x124a
```

**Comment by qdm12** [comment]
@max06 the panic you saw should be fixed in the image tags `:v2.5.0` and `:latest`.

The docs just got fixed (100% credits to @stavros-k) just now.

**Comment by max06** [comment]
@qdm12 glad to see you back in action! Thanks for the fix!

I'll enable the new image here and keep an eye on it 👍🏼
